omod/src/test/java/org/openmrs/module/aijar/htmlformentry/DSDMProgramSubmissionActionTest.java

```diff
@@ -0,0 +1,267 @@
+package org.openmrs.module.aijar.htmlformentry;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.*;
+import org.openmrs.api.ProgramWorkflowService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.htmlformentry.FormEntryContext;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+import org.openmrs.module.htmlformentry.HtmlForm;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpSession;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests patient enrollment into the MCH program
+ */
+public class DSDMProgramSubmissionActionTest extends BaseModuleWebContextSensitiveTest {
```

Comment:
@slubwama For each of the test cases, add checks for program enrollment, and program exit to verify that these dates match what is expected in the database

Reply:
@ssmusoke this is complete.